src/js/helpers/timeline/declarationHelper-nodes.spec.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2020 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getProcessedNodes } from './declarationHelper';
+
+test('One-node-graph is returned as is', () => {
+  const graph = {
+    id: 'A',
+    label: 'A label',
+    status: 'success',
+    type: 'single',
+    response: {
+      invocations: {},
+      transition: '_success',
+    },
+  };
+
+  expect(getProcessedNodes(graph)).toStrictEqual([graph]);
+});
+
+test('Processed child nodes are returned', () => {
+  const child = {
+    id: 'AA',
+    label: 'AA label',
+    status: 'success',
+    type: 'single',
+    response: {
+      invocations: {},
+      transition: '_success',
+    },
+  };
+
+  const graph = {
+    id: 'A',
+    label: 'A label',
+    status: 'success',
+    type: 'single',
+    on: {
+      _success: child,
+    },
+    response: {
+      invocations: {},
+      transition: '_success',
+    },
+  };
+
+  expect(getProcessedNodes(graph)).toStrictEqual([graph, child]);
+});
+
+test('Unpocessed child nodes are not returned', () => {
```

Comment:
What will happen when the root node is `unprocessed` or even `empty`?

Reply:
It'll return an empty array and therefore the timeline will be empty. I can cover this case, but is it even possible for a root node to be unprocessed?